### ISSUE TITLE
Client.closed is false before opening a connection

### ISSUE THREAD

**Issue by ssstuckey** [patrickjuchli/basic-ftp#94]
**Describe the bug**
Initialization of Client.closed property doesn't account for the possibility that it will be checked prior to opening the connection the first time. 

Not sure this is a critical issue, but it caught me out. I have worked around it by keeping my own init state to decide if I've ever opened the connection.

**Example code**
```
const assert = require('assert');
const ftp = require('basic-ftp');
const myClient = new ftp.Client();
assert(myClient.closed, 'Client has not been opened but reports !closed');
```

**Console output**
assert.js:373
    throw err;
    ^

AssertionError [ERR_ASSERTION]: Client has not been opened but reports !closed

**Which version of Node.js are you using?**
Node v12.8.0
basic-ftp@3.8.7

**Comment by patrickjuchli** [comment]
Interesting. So far, `closed` indicated if the client has been closed by the user or by an error as described in the documentation. You are describing something else, something more abstract like the client being `disconnected`. 

But I think your point is valid, the meaning of `closed` before connecting for the first time is not clear. I think it's okay to change the implementation of `closed` to describe being disconnected.

Ideally, we'd also call the property `disconnected` but I don't want to break the API.

**Comment by patrickjuchli** [comment]
Fixed in 4.0.1
